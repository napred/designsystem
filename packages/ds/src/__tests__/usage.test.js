/**
 * @flow
 * @jest-environment jsdom
 */

import React from 'react';
import { render } from 'react-testing-library';
import { Box, DesignSystem } from '../';

function RedBox(props) {
  return <Box {...props} />;
}

RedBox.defaultProps = {
  color: 'red',
};

describe('design system usage', () => {
  it('works correctly with default system stylers', () => {
    // $FlowFixMe
    const { asFragment } = render(
      <DesignSystem>
        <Box>Default Box</Box>
        <Box color="blue">Blue box</Box>
        <Box as="h1" color="green">
          Green title
        </Box>
        <Box as={RedBox} color="green" p={1} m={0} bgColor="black">
          Green RedBox
        </Box>
      </DesignSystem>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
