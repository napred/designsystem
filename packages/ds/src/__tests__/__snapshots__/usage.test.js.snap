// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`design system usage works correctly with default system stylers 1`] = `
<DocumentFragment>
  <div
    class="emotion-0"
  >
    Default Box
  </div>
  .emotion-0 {
  color: #3B5998;
}

<div
    class="emotion-0"
    color="blue"
  >
    Blue box
  </div>
  .emotion-0 {
  color: #3CD048;
}

<h1
    class="emotion-0"
    color="green"
  >
    Green title
  </h1>
  .emotion-0 {
  color: #3CD048;
  padding: 4px;
  margin: 0px;
  background-color: black;
}

<div
    class="emotion-0 emotion-0"
    color="green"
  >
    Green RedBox
  </div>
</DocumentFragment>
`;
